src/com/owncloud/android/ui/activity/PatternLockActivity.java

```diff
@@ -0,0 +1,270 @@
+package com.owncloud.android.ui.activity;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.res.Configuration;
+import android.preference.PreferenceManager;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.andrognito.patternlockview.PatternLockView;
+import com.andrognito.patternlockview.listener.PatternLockViewListener;
+import com.andrognito.patternlockview.utils.PatternLockUtils;
+import com.owncloud.android.BuildConfig;
+import com.owncloud.android.R;
+import com.owncloud.android.lib.common.utils.Log_OC;
+
+import java.util.List;
+
+public class PatternLockActivity extends AppCompatActivity {
+
+    private static final String TAG = PatternLockActivity.class.getSimpleName();
+
+    public final static String PREFERENCE_SET_PATTERN = "set_pattern";
+    public final static String ACTION_REQUEST_WITH_RESULT = "ACTION_REQUEST_WITH_RESULT";
+    public final static String ACTION_CHECK_WITH_RESULT = "ACTION_CHECK_WITH_RESULT";
+    public final static String ACTION_CHECK = "ACTION_CHECK_PATTERN";
+
+    public final static String KEY_PATTERN  = "KEY_PATTERN";
+    public final static String KEY_CHECK_RESULT = "KEY_CHECK_PATTERN_RESULT";
+
+    private static String KEY_CONFIRMING_PATTERN = "CONFIRMING_PATTERN";
+    private static String KEY_PATTERN_STRING = "PATTERN_STRING";
+    private static String PATTERN_HEADER_VIEW_TEXT = "PATTERN_HEADER_VIEW_TEXT";
+    private static String PATTERN_EXP_VIEW_STATE = "PATTERN_EXP_VIEW_STATE";
+    private static String COUNT_VALUE = "COUNT_VALUE";
+
+    private boolean patternPresent = false;
+    private boolean patternExpViewState = false;
```

Comment:
This is only used in onCreate method, so you wouldn't need a global variable

Reply:
@davigonz Actually I am using these variables in many methods not just onCreate() hence they should be global.

Reply:
I cannot see where are patternExpViewState and patternHeaderViewText variables being used, apart from onCreate() method

Reply:
@davigonz Yes I did not check for all the variables that I have declared. I will fix it now.

Reply:
@davigonz Yes I did not check for all the variables that I have declared. I will fix it now.

Reply:
Thanks @sd1998 , whenever you can, no hurries